### PR TITLE
[bitnami/thanos] Creates separate service for grpc port of query module

### PR DESCRIPTION
**Pull request by felippe-mendonca**
Signed-off-by: Felippe Mendonça <mendonca.felippe@gmail.com>

### Description of the change

As previously discussed at https://github.com/bitnami/charts/issues/11028, this PR creates a separate Kubernetes Service for Thanos Query gRPC port. 

### Benefits

Having a separate gRPC service, will be possible to configure this Service with specific annotations in order to make gRPC Ingress works, such as Traefik 2.0.

### Applicable issues

  - fixes https://github.com/bitnami/charts/issues/11028

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/thanos/templates/query/service-grpc-headless.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
+    {{- end }}
+    {{- if .Values.query.serviceGrpc.annotations }}
+    {{- include "common.tplvalues.render" (dict "value" .Values.query.serviceGrpc.annotations "context" $) | nindent 4 }}
+    {{- end }}
+spec:
+  type: ClusterIP
+  clusterIP: None
+  ports:
+    - port: {{ if .Values.query.service.grpc }}{{ coalesce .Values.query.service.ports.grpc .Values.query.service.grpc.port }}{{ else }}{{ .Values.query.service.ports.grpc }}{{ end }}
```
Comment (fmulero) [inline bitnami/thanos/templates/query/service-grpc-headless.yaml:23]: Typo here and I think we can simplify:
```suggestion
    - port: {{ .Values.query.serviceGrpc.ports.grpc }}
```

**Review by fmulero — CHANGES_REQUESTED** [review]
Thanks a lot @felippe-mendonca I looks very well, I have only two little suggestions. Could you have it a look?

**Comment by felippe-mendonca** [comment]
> Thanks a lot @felippe-mendonca I looks very well, I have only two little suggestions. Could you have it a look?

Hey @fmulero! Thanks for your review, I've accept the suggested changes!

File: bitnami/thanos/templates/query/service.yaml [hunk 2]
```diff
@@ -32,7 +32,7 @@ spec:
   loadBalancerSourceRanges: {{- toYaml $query.service.loadBalancerSourceRanges | nindent 4 }}
   {{- end }}
   ports:
-    - port: {{ if $query.service.http }}{{ coalesce $query.service.ports.http $query.service.http.port }}{{ else }}{{ $query.service.ports.http }}{{ end }}
+    - port: {{ $query.service.ports.http }}
```
Comment (felippe-mendonca) [inline bitnami/thanos/templates/query/service.yaml:35]: @fmulero also changed here, removing the backward compatibility config

**Review by fmulero — DISMISSED** [review]
LGTM, Well done!. I am going to run some more tests, if everything goes well your PR will be merge by bitnami-bot.

**Pull request merged**